lightstep/lightstep_b3_propagator.py

```diff
@@ -0,0 +1,127 @@
+        baggage = span_context._baggage
+        if baggage != span.SpanContext.EMPTY_BAGGAGE:
+            parentspanid = baggage.get(self._PARENTSPAN, None)
+            if parentspanid is not None:
+                carrier[_PARENTSPANID] = parentspanid
+            flags = baggage.get(self._FLAGS, None)
+            if flags is not None:
+                carrier[self._FLAGS] = flags
+            sampled = baggage.get(self._SAMPLED, None)
+            if sampled is not None:
+                if flags == 1:
+                    warn(
+                        "x-b3-flags: 1 implies x-b3-sampled: 1, not sending "
+                        "the value of x-b3-sampled"
+                    )
+                else:
+                    carrier[_SAMPLED] = sampled
+                    if sampled in [True, False]:
+                        warn(
+                            "The value of x-b3-sampled should "
+                            "be {} instead of {}".format(
+                                int(sampled), sampled
+                            )
+                        )
+        else:
+            if traceid is None or spanid is None:
+                raise Exception(
+                    "If not propagating only the sampling state, traceid and "
+                    "spanid must be defined"
+                )
+
+    def extract(self, carrier):
+        case_insensitive_carrier = {}
+        for key, value in carrier.items():
+            for b3_key in [
+                _SINGLE_HEADER,
+                _TRACEID,
+                _SPANID,
+                _PARENTSPANID,
+                _SAMPLED,
+                _FLAGS,
+            ]:
+                if re_compile(key, IGNORECASE).match(b3_key):
+                    case_insensitive_carrier[b3_key] = value
+        carrier = case_insensitive_carrier
+        if _SINGLE_HEADER in carrier.keys():
+            fields = carrier[_SINGLE_HEADER].split("-", 4)
+            len_fields = len(fields)
+            if len_fields < 3:
+                if len(fields) == 1:
+                    sampled = fields[0]
+                elif len(fields) == 2:
+                    traceid, spanid = fields
+                baggage = None
+            else:
+                baggage = {}
+                if len(fields) == 3:
+                    traceid, spanid, sampled = fields
+                else:
+                    traceid, spanid, sampled, parent_spanid = fields
+                    baggage[_PARENTSPANID] = parent_spanid
+                if sampled == "d":
+                    baggage[_FLAGS] = 1
+                else:
+                    baggage[_SAMPLED] = sampled
+        else:
+            traceid = carrier.get(_TRACEID, None)
+            spanid = carrier.get(_SPANID, None)
+            parentspanid = carrier.get(_PARENTSPANID, None)
+            sampled = carrier.get(_SAMPLED, None)
+            flags = carrier.get(_FLAGS, None)
+            if parentspanid is sampled is flags is None:
```

Comment:
`if parentspanid is sampled` doesn't look/read right. Could you confirm this is the expected check?

Reply:
Yes, I'm checking that all three of `parentspanid`, `sampled` and `flags` are `None`. I am using `is` instead of `==` because `None` [is a singleton](https://www.python.org/dev/peps/pep-0008/#programming-recommendations).